File: Core.DomainModel/ItSystem/ItSystemUsage.cs
using System.Collections.Generic;

namespace Core.DomainModel.ItSystem
{
    public class ItSystemUsage : Entity, IHasRights<ItSystemRight>
    {
        public ItSystemUsage()
        {
            this.Contracts = new List<ItContract.ItContract>();
            this.Wishes = new List<Wish>();
            this.OrgUnits = new List<OrganizationUnit>();
            this.TaskRefs = new List<TaskRef>();
            this.Rights = new List<ItSystemRight>();
            this.InterfaceUsages = new List<InterfaceUsage>();
            this.InterfaceExposures = new List<InterfaceExposure>();
            this.UsedBy = new List<OrganizationUnit>();
            this.ItProjects = new List<ItProject.ItProject>();
        }
        
        public bool IsStatusActive { get; set; }
        public string Note { get; set; }
        public string LocalSystemId { get; set; }
        
        public string EsdhRef { get; set; }
        public string CmdbRef { get; set; }
        public string DirectoryOrUrlRef { get; set; }
        public string AdOrIdmRef { get; set; }

        public int? ResponsibleUnitId { get; set; }
        public virtual OrganizationUnit ResponsibleUnit { get; set; }
        
        public int OrganizationId { get; set; }
        public virtual Organization Organization { get; set; }
        
        public int ItSystemId { get; set; }
        public virtual ItSystem ItSystem { get; set; }

        public int? ArchiveTypeId { get; set; }
        public virtual ArchiveType ArchiveType { get; set; }

        public int? SensitiveDataTypeId { get; set; }
        public virtual SensitiveDataType SensitiveDataType { get; set; }

        public int? OverviewItSystemId { get; set; }
        public virtual ItSystem OverviewItSystem { get; set; }

        public virtual ICollection<ItContract.ItContract> Contracts { get; set; }
        public virtual ICollection<Wish> Wishes { get; set; }
        public virtual ICollection<OrganizationUnit> OrgUnits { get; set; }

        /// <summary>
        /// These Organization Units are using this system
        /// </summary>
        public virtual ICollection<OrganizationUnit> UsedBy { get; set; }
        
        /// <summary>
        /// IT System support these tasks
        /// </summary>
        public virtual ICollection<TaskRef> TaskRefs { get; set; }

        public virtual ICollection<ItSystemRight> Rights { get; set; }

        /// <summary>
        /// The local usages of interfaces. 
        /// </summary>
        public virtual ICollection<InterfaceUsage> InterfaceUsages { get; set; }

        /// <summary>
        /// The local exposures of interfaces.
        /// </summary>
        public virtual ICollection<InterfaceExposure> InterfaceExposures { get; set; }

        public virtual ICollection<ItProject.ItProject> ItProjects { get; set; }
    }
}

File: Presentation.Web/app/models/it-system-usage/it-system-usage.ts
﻿module Kitos.Models.ItSystemUsage {
    import ItSystem = Models.ItSystem.IItSystem;

    export interface IItSystemUsage extends IEntity {
        /** Gets or sets the note. */
        Note: string;
        /** Gets or sets the user defined local system identifier. */
        LocalSystemId: string;
        /** Gets or sets the version. */
        Version: string;
        /** Gets or sets a reference to relevant documents in an extern CMDB system. */
        CmdbRef: string;
        /** Gets or sets the local call system. */
        LocalCallName: string;
        /** Organization Unit responsible for this system usage. */
        ResponsibleUsage: IItSystemUsageOrgUnitUsage;
        OrganizationId: number;
        /** Gets or sets the organization marked as responsible for this it system usage. */
        Organization: IOrganization;
        ItSystemId: number;
        /** Gets or sets the it system this instance is using. */
        ItSystem: ItSystem;
        ArchiveTypeId: number;
        ArchiveType: ItSystem.IArchiveType;

        ArchiveLocationId: number;
        ArchiveLocation: ItSystem.IArchiveLocation;

        ArchiveTestLocationId: number;
        ArchiveTestLocation: ItSystem.IArchiveTestLocation;

        SensitiveDataTypeId: number;
        SensitiveDataType: ItSystem.ISensitiveDataType;
        OverviewId: number;
        /** Gets or sets the it system usage that is set to be displayed on the it system overview page. */
        Overview: IItSystemUsage;
        /** Gets or sets the main it contract for this instance.The it contract is used to determine whether this instanceis marked as active/inactive. */
        MainContract: ItContract.IItContractItSystemUsage;
        /** Gets or sets it contracts associated with this instance. */
        Contracts: Array<ItContract.IItContractItSystemUsage>;
        /** Gets or sets the organization units that are using this instance. */
        UsedBy: Array<IItSystemUsageOrgUnitUsage>;
        /** Gets or sets the tasks this instance supports. */
        TaskRefs: Array<ITaskRef>;
        /** Gets or sets the associated it projects. */
        ItProjects: Array<ItProject.IItProject>;

        ArchivePeriods: ItSystem.IArchivePeriode[];

        ReferenceId: number;
        Reference: IReference;

        Concluded?: Date;
        ArchiveDuty: string;
        Registertype: boolean;

        IsActive: boolean;

        RiskSupervisionDocumentationUrl: string;
        RiskSupervisionDocumentationUrlName: string;
        LinkToDirectoryUrl: string;
        LinkToDirectoryUrlName: string;

        SensitiveDataLevels: Models.Odata.ItSystemUsage.IItSystemUsageSensitiveDataLevelDTO[];

        AssociatedDataProcessingRegistrations: Models.OData.DataProcessing.IDataProcessingRegistration[];
    }
}
